whelktool/src/main/groovy/whelk/datatool/form/Transform.groovy

```diff
@@ -420,6 +298,30 @@ class Transform {
         return nodeIdMappings
     }
 
+    Map<String, Set<String>> collectBaseTypeMappings(JsonLd jsonLd) {
+        Map<String, Set<String>> mappings = [:]
+
+        if (jsonLd == null) {
+            return mappings
+        }
+
+        DocumentUtil.traverse(matchForm) { node, path ->
+            if (node instanceof Map && node.containsKey(_MATCH) && ((List) node[_MATCH]).contains(SUBTYPES)) {
+                def baseType = (String) node[TYPE_KEY]
+                Set<String> subTypes = getSubtypes(baseType, jsonLd) as Set
+                mappings[baseType] = subTypes
+                return new DocumentUtil.Nop()
+            }
+        }
+
+        return mappings
+    }
+
+    @Memoized
```

Comment:
This will give us trouble if we ever want to be able to refresh vocab without restarting whelk.
I think it would be better to reuse the Transform object within one bulk change, i.e. pass it to modify() inside the whelktool script.
But we can leave it as is for now.